packages/visual-editor/src/internal/components/InternalLayoutEditor.tsx

```diff
@@ -169,11 +173,11 @@ export const InternalLayoutEditor = ({
         alert("Failed to paste: Invalid component data.");
         return;
       }
-      // Update the component data in the specified zone
+
       dispatch({
```

Comment:
We can get the data with updated ids with walkTree. Btw, I had a DM with someone who took this code and updated things to work with 0.19.x which is where I'm getting this.

```
// If the pasted data has children, we need to generate new ids for them
      const newData = walkTree(pastedData, baseConfig, (contents) => (
        contents.map((item: ComponentDataOptionalId) => {
          const id = `${item.type}-${uuidv4()}`;
          return {
            ...item,
            props: { ...item.props, id },
          };
        })
      ))
```